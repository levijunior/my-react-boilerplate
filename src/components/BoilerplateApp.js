import React, { useState, useEffect, useRef } from 'react';

import { FormContainer, InputEl, ButtonEl } from './styled-components/form'
import { H1El, H2El } from './styled-components/text'

import Connect from "../store/config/connect";
import { __boilerplateAction } from "../store/__boilerplateDuck"; 


const App = props => {

  const inputEl = useRef(null);

  const [__boilerplate, setBoilerplate] = useState("");
  const changeNameHandler = ({ target: { value } }) => setBoilerplate(value);
  const onSubmitHandler = e => {
    e.preventDefault()
    props.dispatch(__boilerplateAction(__boilerplate))
  };

  useEffect(() => {
    inputEl.current.focus();
  }, []);

  return (
    <>
      <H1El>{props.__boilerplate}</H1El>
      <FormContainer onSubmit={onSubmitHandler}>
        <InputEl
          value={__boilerplate}
          onChange={changeNameHandler}
          type="text"
          placeholder="__boilerplate"
          ref={inputEl} 
          data-testid="inputEl"
        />
        <ButtonEl
          data-testid="buttonEl"
        >
          change store state
        </ButtonEl>
      </FormContainer>
      <H2El data-testid="h2El">{__boilerplate}</H2El>
    </>
  );
};

const mapStateToProps = ({ data }) => {
  return {
    __boilerplate: data
  };
};

export default Connect(mapStateToProps)(App);

