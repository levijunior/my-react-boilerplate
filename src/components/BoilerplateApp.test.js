import React from 'react'
import { render, waitForElement, fireEvent, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import BoilerplateApp from './BoilerplateApp'

afterEach(cleanup)

it('renders and compare to snapshot', () => {
  const { asFragment } = render(<BoilerplateApp />)
  expect(asFragment()).toMatchSnapshot()
});

describe('Test example for __boilerplate',  () => {
  it('Update the __boilerplateDuck store', async () => {
    const { getByTestId } = render( <BoilerplateApp /> )

    const fieldNode = await waitForElement( () => getByTestId('inputEl') )
    const btnNode = await waitForElement( () => getByTestId('buttonEl') )
    const h1Node = await waitForElement( () => getByTestId('h2El') )

    const testText = 'test text'
    fireEvent.change(fieldNode, { target: {value: testText} })
    expect(fieldNode.value).toEqual(testText)

    fireEvent.click(btnNode)
    
    expect(h1Node).toHaveTextContent(testText)

  })
})
