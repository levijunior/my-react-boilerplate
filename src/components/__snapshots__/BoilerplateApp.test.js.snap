// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and compare to snapshot 1`] = `
<DocumentFragment>
  <h1
    class="sc-bxivhb bTwjlV"
  >
    __BOILERPLATE
  </h1>
  <div
    class="sc-bdVaJa bhVcIp"
  >
    <input
      class="sc-bwzfXH fGpkfM"
      data-testid="inputEl"
      placeholder="__boilerplate"
      type="text"
      value=""
    />
    <button
      class="sc-htpNat eQEgLM"
      data-testid="buttonEl"
    >
      change store state
    </button>
  </div>
  <h2
    class="sc-ifAKCX bUaWFB"
    data-testid="h2El"
  />
</DocumentFragment>
`;
